src/shared/Atlassian.Bitbucket/BitbucketTokenEndpointResponseJson.cs

```diff
@@ -1,12 +1,57 @@
+using System;
+using System.Text.Json;
 using GitCredentialManager.Authentication.OAuth.Json;
-using Newtonsoft.Json;
+using System.Text.Json.Serialization;
 
-namespace Atlassian.Bitbucket 
+namespace Atlassian.Bitbucket
 {
-        public class BitbucketTokenEndpointResponseJson : TokenEndpointResponseJson
+    [JsonConverter(typeof(BitbucketCustomTokenEndpointResponseJsonConverter))]
+    public class BitbucketTokenEndpointResponseJson : TokenEndpointResponseJson
+    {
+        // To ensure the "scopes" property used by Bitbucket is deserialized successfully with System.Text.Json, we must
+        // use a custom converter. More information can be found here:
+        // https://learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/converters-how-to?pivots=dotnet-6-0#support-polymorphic-deserialization
+    }
+
+    public class BitbucketCustomTokenEndpointResponseJsonConverter : JsonConverter<BitbucketTokenEndpointResponseJson>
+    {
+        public override BitbucketTokenEndpointResponseJson Read(
+            ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
         {
-            // Bitbucket uses "scopes" for the scopes property name rather than the standard "scope" name
-            [JsonProperty("scopes")]
-            public override string Scope { get; set; }
+            if (reader.TokenType != JsonTokenType.StartObject)
+            {
+                throw new JsonException();
+            }
+
+            var response = new BitbucketTokenEndpointResponseJson();
+
+            while (reader.Read())
+            {
+                if (reader.TokenType == JsonTokenType.EndObject)
+                {
+                    return response;
+                }
+
+                if (reader.TokenType == JsonTokenType.PropertyName)
+                {
+                    var propertyName = reader.GetString();
+                    reader.Read();
+                    if (propertyName != null)
+                    {
+                        // Bitbucket uses "scopes" for the scopes property name rather than the standard "scope" name
+                        if (propertyName.Equals("scopes", StringComparison.OrdinalIgnoreCase))
```

Comment:
How are the other properties on the JSON response object populated by this converter?